File: array/rotateimage.cpp
class Solution{
    public:
    void rotate(vector<vector<int>>& m)
    {
    //         on2 it was than we haad to optimze
//         hence transpose
//         reverse the row in order for the rseult 
//         space o1
//         tc on2+on2

        int n = m.size();
        
        for(int i=0; i<n; i++)
        
            for(int j =0; j<i; j++)
                swap(m[i][j], m[j][i]);
            
            
        for(int i=0; i<n; i++)
       reverse(m[i].begin(), m[i].end());
        
    }
};

File: array/rotate image.rs
impl Solution {
    pub fn rotate(matrix: &mut Vec<Vec<i32>>) {
            let mut matrix = matrix;
        
        let (len, n) = (matrix.len(), matrix.len() - 1);
        
        for x in 0..len / 2 {
            for y in 0..(len + 1) / 2 {
                
                let temp = matrix[x][y];
                matrix[x][y] = matrix[n - y][x];
                matrix[n - y][x] = matrix[n - x][n - y];
                matrix[n - x][n - y] = matrix[y][n - x];
                matrix[y][n - x] = temp;
            }
        }
    }
}
